pkg/phlare/modules.go

```diff
@@ -135,14 +135,22 @@ func (f *Phlare) initRuntimeConfig() (services.Service, error) {
 }
 
 func (f *Phlare) initTenantSettings() (services.Service, error) {
-	mem, err := settings.NewMemoryStore()
+	var store settings.Store
+	var err error
+
+	switch {
+	case f.storageBucket != nil:
+		store, err = settings.NewBucketStore(f.storageBucket)
+	default:
+		store, err = settings.NewMemoryStore()
```

Comment:
A warning that settings are not stored would be nice

Reply:
Good idea, added in https://github.com/grafana/pyroscope/pull/2905/commits/6274c9877a39ed55190464aa7a7217a039403f35